presto-spi/src/main/java/io/prestosql/spi/type/SqlTimestamp.java

```diff
@@ -43,7 +45,26 @@ public static SqlTimestamp fromMillis(int precision, long millis)
 
     public static SqlTimestamp newInstance(int precision, long epochMicros, int picosOfMicro)
     {
-        return newInstanceWithRounding(precision, epochMicros, picosOfMicro);
+        if (precision <= 6) {
```

Comment:
should the `fromMillis` above do rounding or checks?

or maybe we (re)move it, since it is used in tests only

Reply:
It's supposed to be a convenience method to instantiate a SqlTimestamp of a given precision, including any necessary rounding. The issue with (re)moving it is that it would require making `newInstanceWithRounding` public, and that's not something I want to be visible.

Reply:
> The issue with (re)moving it is that it would require making newInstanceWithRounding public, and that's not something I want to be visible.

`fromMillis` currently doesn't call `newInstanceWithRounding`, it calls `newInstance`

(perhaps incorectly so?)

Reply:
Oh yeah, I fixed that in my branch and forgot to push :)